src/controllers/fxaRpEvents.js

```diff
@@ -0,0 +1,167 @@
+  const headerToken = authHeader.substr(authHeader.indexOf(' ') + 1)
+
+  // Decode the token, require it to come out ok as an object
+  const token = jwt.decode(headerToken, { complete: true })
+  if (!token || typeof token === 'string') {
+    throw UnauthorizedError('Invalid token type')
+  }
+
+  // Verify we have a key for this kid, this assumes that you have fetched
+  // the publicJwks from FxA and put both them in an Array.
+  const publicJwks = await getJwtPubKey()
+  const jwk = publicJwks.find(j => j.kid === token.header.kid)
+  if (!jwk) {
+    throw UnauthorizedError('No jwk found for this kid: ' + token.header.kid)
+  }
+  const jwkPem = jwkToPem(jwk)
+
+  // Verify the token is valid
+  const decoded = jwt.verify(headerToken, jwkPem, {
+    algorithms: ['RS256']
+  })
+  // This is the JWT data itself.
+  return decoded
+}
+
+/**
+ * Handler for FxA events, used by FxA as a callback URI endpoint
+ * Example events include FxA user deletion, profile changes, and subscription changes
+ *
+ * @param {*} req
+ * @param {*} res
+ * @returns
+ */
+const fxaRpEvents = async (req, res) => {
+  let decodedJWT
+  try {
+    decodedJWT = await authenticateFxaJWT(req)
+  } catch (e) {
+    log.error('fxaRpEvents', e)
+    captureException(e)
+    res.status(202)
+  }
+
+  if (!decodedJWT?.events) {
+    // capture an exception in Sentry only. Throwing error will trigger FXA retry
+    log.error('fxaRpEvents', decodedJWT)
+    captureException(new Error('fxaRpEvents: decodedJWT is missing attribute "events"', decodedJWT))
+    res.status(202)
+  }
+
+  const fxaUserId = decodedJWT?.sub
+  if (!fxaUserId) {
+    // capture an exception in Sentry only. Throwing error will trigger FXA retry
+    captureException(new Error('fxaRpEvents: decodedJWT is missing attribute "sub"', decodedJWT))
+    res.status(202)
+  }
+
+  const subscriber = getSubscriberByFxaUid(fxaUserId)
+
+  // reference example events: https://github.com/mozilla/fxa/blob/main/packages/fxa-event-broker/README.md
+  for (const event in decodedJWT?.events) {
+    switch (event) {
+      case FXA_DELETE_USER_EVENT:
+        log.debug('fxa_delete_user', {
+          event
+        })
+
+        // delete user events only have keys. Keys point to empty objects
+        await deleteSubscriberByFxAUID(fxaUserId)
+        break
+      case FXA_PROFILE_CHANGE_EVENT: {
+        const updatedProfileFromEvent = decodedJWT.events[event]
+        log.debug('fxa_profile_update', {
+          event,
+          updatedProfileFromEvent
+        })
+
+        // get current profiledata
+        const { fxa_profile_json: currentFxAProfile } = subscriber
+
+        // merge new event into existing profile data
+        for (const key in updatedProfileFromEvent) {
+          if (currentFxAProfile[key]) currentFxAProfile[key] = updatedProfileFromEvent[key]
+        }
+
+        // update fxa profile data
+        await updateFxAProfileData(subscriber, currentFxAProfile)
```

Comment:
`tsc` complains about this `await` but that's because the JSDoc on the callee is wrong :)